src/components/Sidebar.js

```diff
@@ -0,0 +1,82 @@
+import React, { Fragment } from 'react';
+import { Link } from 'gatsby';
+
+// Toggles sidebar menus, only one may be open at a time
+function toggleMenu(id) {
+  const element = document.getElementById(id);
+
+  const scanners = document.getElementById('Scanners');
+  const persistenceProviders = document.getElementById('Persistence Providers');
+  const hooks = document.getElementById('Hooks');
+
+  if (element !== scanners) {
+    scanners.style.display = 'none';
+  }
+  if (element !== persistenceProviders) {
+    persistenceProviders.style.display = 'none';
+  }
+  if (element !== hooks) {
+    hooks.style.display = 'none';
+  }
+
+  element.style.display === 'block'
+    ? (element.style.display = 'none')
+    : (element.style.display = 'block');
+}
+
+// Determine the naming of Sidebar categories
+function getCategoryTitle(category) {
+  let categoryTitle = '';
+
+  switch (category[0].node.frontmatter.category) {
+    case 'scanner':
+      categoryTitle = 'Scanners';
+      break;
+
+    case 'hook':
+      category[0].node.frontmatter.type === 'persistenceProvider'
+        ? (categoryTitle = 'Persistence Providers')
+        : (categoryTitle = 'Hooks');
+      break;
+
+    default:
+      categoryTitle = 'Unknown Category';
+      break;
+  }
+
+  return categoryTitle;
+}
+
+const Sidebar = (props) => {
+  return (
+    <nav className="sidebar">
+      {props.dataArray.map((category, index) => (
+        <Fragment key={index}>
+          <h1
```

Comment:
ESLint is Outputting a11y warnings:

```
warn ESLintError:
/Users/jannikhollenbach/src/securecodebox.github.io/src/components/Sidebar.js
  55:11  warning  Visible, non-interactive elements with click handlers must have at least one keyboard listener  jsx-a11y/click-events-have-key-events
  55:11  warning  Non-interactive elements should not be assigned mouse or keyboard event listeners               jsx-a11y/no-noninteractive-element-interactions
```

These should be evaluated or explicitly disabled for the offending line if they are false-positives.